receiver/mysqlreceiver/client.go

```diff
@@ -164,13 +164,20 @@ type ReplicaStatusStats struct {
 var _ client = (*mySQLClient)(nil)
 
 func newMySQLClient(conf *Config) client {
+	tls, err := conf.TLS.LoadTLSConfig()
+	tlsConfig := ""
+	if err == nil && tls != nil && mysql.RegisterTLSConfig("custom", tls) == nil {
+		tlsConfig = "custom"
+	}
```

Comment:
`if tls == nil` we should skip this.

Reply:
Agreed, I am only setting `tlsConfig` if `tls != nil`